### PR TITLE
fix: ensure drained() resolves after async tasks complete

### LINKED ISSUE #88: not waiting for async tasks to finish before resolving `drained()`
The drain() callback seems to work correctly, however the promise returned by drained() gets resolved before any async work has been done.

Example:

```js
const fastq = require('fastq');

(async () => {
    async function prosessTask() {
      await new Promise(resolve => setTimeout(resolve, 0));
      console.log('processed');
    }
    
    try{
      const queue = fastq.promise(prosessTask, 1); // run sequentially
      queue.drain = () => console.log('called drain');
      queue.drained().then( () => console.log('drained promise resolved') );
    
      await Promise.all([queue.push(), queue.push(), queue.push()]);
    }catch(err){
        console.error(err);
    }
})();

```

The output logs I'm getting look like this:

```
drained promise resolved
processed
processed
processed
called drain
```

I would have expected:
```
processed
processed
processed
called drain
drained promise resolved
```

### PR DESCRIPTION
**Pull request by todoroff**
fixes #88

### CODE REVIEW

File: queue.js [hunk 1]
```diff
@@ -288,19 +288,19 @@ function queueAsPromised (context, worker, _concurrency) {
...
-    }
-
-    var previousDrain = queue.drain
-
     var p = new Promise(function (resolve) {
-      queue.drain = function () {
-        previousDrain()
-        resolve()
-      }
+      setImmediate(() => {
```
Comment (mcollina) [inline queue.js:292]: Could you check if a queueMicroTask() call does the job as well? I would prefer not to turn the event loop.

Comment (todoroff) [reply queue.js:292]: > Could you check if a queueMicroTask() call does the job as well? I would prefer not to turn the event loop.

That won't run on older versions of node though and I noticed in other PRs that you want it to be compatible all the way back essentially.

process.nextTick() does the job and is even higher priority than queueMicroTask(), so I can use that if you like? My worry was adding the ability to potentially starve the event loop, although it shouldn't be possible with normal usage.

Let me know how you'd like to proceed.

Comment (mcollina) [reply queue.js:292]: `nextTick` is perfect here, thanks!

**Review by mcollina — APPROVED** [review]
lgtm

File: queue.js (hunk 1 above)
Comment (summerkiflain) [reply queue.js]: @mcollina @todoroff I was using this library in my react-native app, unfortunately q.drained() doesn't get called anymore after migrating to latest version, can we add something that works well with the other environments also, In my case process.nextTick doesn't get called in react-native.

**Pull request merged**